repose-aggregator/components/filters/herp-filter/src/main/scala/org/openrepose/filters/herp/HerpFilter.scala

```diff
@@ -121,8 +123,17 @@ class HerpFilter @Inject()(configurationService: ConfigurationService,
       else stripHeaderParams(allProjectIds.maxBy(getQuality))
     }
 
-    val projectIds = httpServletRequest.getHeaders(OpenStackServiceHeader.TENANT_ID.toString).asScala
-      .++(httpServletRequest.getHeaders(X_PROJECT_ID).asScala).toTraversable
+    def getSplitHeaders(headerInteractor: HeaderInteractor, headers: List[String]): Traversable[String] = {
+      headers.foldLeft(Traversable[String]()) { (accumulator, current) =>
+        accumulator.++(headerInteractor.getSplittableHeaders(current).asScala)
+      }
+    }
+
+    val tenantProjectHeaders = List(OpenStackServiceHeader.TENANT_ID.toString, X_PROJECT_ID)
+    val projectIds = getSplitHeaders(new HttpServletRequestWrapper(httpServletRequest), tenantProjectHeaders) match {
+      case x if x.nonEmpty => x
+      case _ => getSplitHeaders(httpServletResponse, tenantProjectHeaders)
+    }
```

Comment:
Now that this is much more manageable, we probably could drop the match for a simple if/else conditional. Again, that's probably a preference thing, but something about using pattern matching to evaluate a conditional expression feels off to me.

```
val reqProjectIds = getSplitHeaders(new HttpServletRequestWrapper(httpServletRequest), tenantProjectHeaders)
val projectIds = if (reqProjectIds.nonEmpty) reqProjectIds else getSplitHeaders(httpServletResponse, tenantProjectHeaders)
```